src/DDTrace/Integrations/Nette/NetteIntegration.php

```diff
@@ -0,0 +1,176 @@
+                $integration->load();
+            });
+        } else {
+            $this->load();
+        }
+
+        return self::LOADED;
+    }
+
+    public function load()
+    {
+        $scope = GlobalTracer::get()->getRootScope();
+        if (!$scope instanceof Scope) {
+            return;
+        }
+
+        $service = \ddtrace_config_app_name(NetteIntegration::NAME);
+
+        $root = $scope->getSpan();
+        $root->setTraceAnalyticsCandidate();
+        $root->overwriteOperationName('nette.request');
+        $root->setTag(Tag::SERVICE_NAME, $service);
+
+        \DDTrace\trace_method(
+            'Nette\Configurator',
+            'createContainer',
+            function (SpanData $span) use ($service) {
+                $span->name = 'nette.configurator.createContainer';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Nette\Configurator',
+            'createRobotLoader',
+            function (SpanData $span) use ($service) {
+                $span->name = 'nette.configurator.createRobotLoader';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Nette\Application\Application',
+            'run',
+            function (SpanData $span) use ($root, $service) {
+                $span->name = 'nette.application.run';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+                $root->setTag(Tag::HTTP_STATUS_CODE, http_response_code());
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Nette\Application\UI\Presenter',
+            'run',
+            function (SpanData $span, $args) use ($root, $service) {
+
+                $span->name = 'nette.presenter.run';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+
+                if (count($args) < 1 || !\is_a($args[0], '\Nette\Application\Request')) {
+                    return;
+                }
+
+                $request = $args[0];
+                $presenter = $request->getPresenterName();
+                $action = $request->getParameter('action');
+
+                $root->setTag(Tag::HTTP_METHOD, $request->getMethod());
+                $root->setTag('nette.route.presenter', $presenter);
+                $root->setTag('nette.route.action', $action);
+                $root->setResource($presenter . ':' . $action);
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Nette\Application\Routers\RouteList',
+            'match',
+            function (SpanData $span) use ($service) {
+                $span->name = 'nette.router.match';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+            }
+        );
```

Comment:
Is there a concept of "route name" or something we could attach here?

Reply:
Not really, the match function on the route list just matches the http request object itself, again this was just ported over from the original PR.

Reply:
@SammyK  What about dropping this one as well? No meta-data, including resource.

Reply:
This one calls the parent `Nette\Routing\RouteList::match()` and [I don't see how it would ever be "slow"](https://github.com/nette/routing/blob/c714f20/src/Routing/RouteList.php#L44-L72) per-se so I think this one could probably be removed.

Reply:
I'll remove it then since there seems to be some consensus.